### DISCUSSION TITLE
Module 4.1 "geodatasets" not being found

### DISCUSSION THREAD

**Discussion by jenw-ua** [ua-gist604b-f24/syllabus#9] [General]
Hello. I know it's a little early for Module 4.1, but I'm leaving town so just trying to stay on top of it. Any help is appreciated!

In Module 4.1, at the point where our instructions say to copy and paste the first set of instructions from the tutorial and into codespace and run it, our module instructions code and the code in the tutorial are different, and neither one is working for me. I'm getting errors with both codes. One saying that geodatasets was deprecated and removed from geopandas and the other saying ModuleNotFoundError: No module named 'geodatasets'.

The code in the instructions says:

import geopandas

path_to_data = geopandas.datasets.get_path("nybb")
gdf = geopandas.read_file(path_to_data)

gdf

The code in the tutorial says:

import geopandas
from geodatasets import get_path

path_to_data = get_path("nybb")
gdf = geopandas.read_file(path_to_data)

gdf

Based on the first code error feedback, it looks like it can be fixed if we get new data located somewhere else? I'm tempted to try and fix it to make this module work, but I thought I'd ask what we should do first, or am I doing something wrong?

Thanks.

Here's two screenshots, one after running each of the two different codes...

<img width="1930" alt="Screenshot 2024-10-29 at 9 58 41 AM" src="https://github.com/user-attachments/assets/d036d984-f279-45e2-b1dc-e2181749483d">

<img width="1177" alt="Screenshot 2024-10-29 at 9 38 45 AM" src="https://github.com/user-attachments/assets/fe804013-6a25-44e2-a1f3-9ef840eaaded">

**Comment by jenw-ua** [comment]
After reading and trying more, it seems to be solved by running this in the terminal window:

pip install geodatasets

Then changing the tutorial's double quotes around nybb to single quotes, like below, to assignment.ipynb:

import geopandas
from geodatasets import get_path

path_to_data = get_path('nybb')
gdf = geopandas.read_file(path_to_data)

gdf

**Reply by ryangrammer** [reply]
thank you Jen for sharing!

**Reply by aaronbirr** [reply]
Thank you

**Reply by aaryno** [reply]
geodatasets was deprecated. The tutorial has the updated directions but the assignment notebook is outdated. We also covered this in the lecture recording.

**Comment by jenw-ua** [comment]
Also, when you get to tutorial #10 and you need to run: gdf.explore("area", legend=False)

You'll probably get an error (at least, I did). If you read the error fine print, it will tell you to run: pip install folium matplotlib mapclassify (run that in your terminal window)...then rerun item 10 in your assignment.ipynb and it will work.
